pytensor/xtensor/shape.py

```diff
@@ -380,3 +383,115 @@ def squeeze(x, dim=None):
         return x  # no-op if nothing to squeeze
 
     return Squeeze(dims=dims)(x)
+
+
+class ExpandDims(XOp):
+    """Add a new dimension to an XTensorVariable."""
+
+    __props__ = ("dim",)
+
+    def __init__(self, dim):
+        self.dim = dim
+
+    def make_node(self, x, size):
+        x = as_xtensor(x)
+
+        if not isinstance(self.dim, str):
+            raise TypeError(f"`dim` must be a string or None, got: {type(self.dim)}")
+
+        if self.dim in x.type.dims:
+            raise ValueError(f"Dimension {self.dim} already exists in {x.type.dims}")
+        if isinstance(size, int | np.integer):
+            if size <= 0:
+                raise ValueError(f"size must be positive, got: {size}")
+        elif not (
+            hasattr(size, "ndim")
+            and getattr(size, "ndim", None) == 0  # symbolic scalar
+        ):
+            raise TypeError(
+                f"size must be an int or scalar variable, got: {type(size)}"
+            )
+
+        # Convert size to tensor
+        size = as_tensor(size, ndim=0)
```

Comment:
Better to convert to as_xtensor, since both tensor and xtensor scalars will be accepted then.

Instead of having the check for numpy specifically, check if after conversion you have `isinstance(size, Constant) and (size.data <= 0)` and then raise.

Are you sure xarray doesn't support size of zero? It's valid otherwise in numpy